Scripts/SecretServer/Coveo/Discovery/Coveo Discovery.ps1

```diff
@@ -0,0 +1,149 @@
+# Expected Argd args=@("Discovery Mode Advanced/Default", "Coveo Tenenant Base URL", "Coveo OrgID",  "API Key","API KeyId","[AdminGroup]=[AdminGroupId]{,[AdditionalGroup]=[AdditionalGroupID]}","[ServiceGroup]=[ServiceGroupId]{,[AdditionalGroup]=[AdditionalGroupID]}"  )
+
+## This block will be used when passing args in from Secret Server
+
+ 
+    $baseURL           = $args[0]
+    $OrgId             = $args[1]
+    $apiKey            = $args[2]
+    $adminUsers        = $args[3]
+    $serviceUsers      = $args[4]
+
+#region define variables
+#Define descriptive variables for use in script & set values.
+
+#Script Constants
+#[string]$LogFile              = "$env:ProgramFiles\Thycotic Software Ltd\Distributed Engine\log\Coveo-Connector.log"
+[string]$LogFile              = "c:\temp\Coveo-Connector.log"
+[int32]$LogLevel              = 3
+[string]$logApplicationHeader = "Coveo Discovery"
+
+$foundAccounts                = @()
+$authHeader                   = @{
+    "Authorization" = "Bearer $ApiKey"
+    "Accqept" = "application/json"
+    "Content-Type" = "application/json"
+}
+#endregion
+
+
+function Write-Log {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateSet(0,1,2,3)]
+        [Int32]$ErrorLevel,
+        [Parameter(Mandatory,ValueFromPipeline)]
+        [string]$Message
+    )
+    # Evaluate Log Level based on global configuration
+    if ($ErrorLevel -le $LogLevel) {
+        # Format message
+        [string]$Timestamp = Get-Date -Format "yyyy-MM-ddThh:mm:sszzz"
+        switch ($ErrorLevel) {
+            "0" { [string]$MessageLevel = "INF0 " }
+            "1" { [string]$MessageLevel = "WARN " }
+            "2" { [string]$MessageLevel = "ERROR" }
+            "3" { [string]$MessageLevel = "DEBUG" }
+        }
+        # Write Log data
+        $MessageString = "{0}`t| {1}`t| {2}`t| {3}" -f $Timestamp, $MessageLevel,$logApplicationHeader, $Message
+        $MessageString | Out-File -FilePath $LogFile -Encoding utf8 -Append -ErrorAction SilentlyContinue
+        # $Color = @{ 0 = 'Green'; 1 = 'Cyan'; 2 = 'Yellow'; 3 = 'Red'}
+        # Write-Host -ForegroundColor $Color[$ErrorLevel] -Object ( $DateTime + $Message)
```

Comment:
@Delinea-RickRoca why is this comment out block left here?

Reply:
Used for debugging when running scripts manually. Not used in automated Logging.  Will ad comments to indicate that in futre releases and will make a note to go back and review once we hit the 34